File: Backend/src/main/java/com/example/demo/Models/GroupModel.java
package com.example.demo.Models;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import com.example.demo.DemoApplication;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.core.JsonProcessingException;


public class GroupModel {

    private long id;
    private String name;
    private String imgPath;
    private String description;
    private UserModel admin;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private Date createdAt;
    private List<PostModel> postModels;
    private List<UserModel> members;


    public GroupModel(){
        id = 0;
        name = "alaa";
        imgPath = "c:\\sdfssgjznfSHJVbhbbagxyj";
        description = "this is the first group in my data base";
        createdAt = new Date();
        members = new ArrayList<>();
    }


    public GroupModel(long id, String name, String imgPath, String description, Date createdAt, List<PostModel> postModels, UserModel admin, List<UserModel> members) {
        this.id = id;
        this.name = name;
        this.imgPath = imgPath;
        this.description = description;
        this.createdAt = createdAt;
        this.postModels = postModels;
        this.admin = admin;
        this.members = members;
    }


    public List<PostModel> getPostModels() {
        return this.postModels;
    }

    public void setPostModels(List<PostModel> postModels) {
        this.postModels = postModels;
    }

    public UserModel getAdmin() {
        return this.admin;
    }

    public void setAdmin(UserModel admin) {
        this.admin = admin;
    }

    public List<UserModel> getMembers() {
        return this.members;
    }

    public void setMembers(List<UserModel> members) {
        this.members = members;
    }

    public long getId() {
        return this.id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getImgPath() {
        return this.imgPath;
    }

    public void setImgPath(String imgPath) {
        this.imgPath = imgPath;
    }

    public String getDescription() {
        return this.description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getCreatedAt() {
        return this.createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    @Override
    public String toString() {
        try {
            return DemoApplication.objectMapper.writeValueAsString(this);
        } catch (JsonProcessingException e) {
            e.printStackTrace();
        }
        return null;
    }

    
    
}

File: Frontend/lib/Model/GroupModel.dart
import 'dart:core';
import 'package:frontend/Model/UserModel.dart';
import 'package:frontend/main.dart';

import 'PostModel.dart';

class GroupModel {
  String? id;
  String? name;
  String? imgPath;
  String? description;
  UserModel? admin;
  DateTime? createdAt;
  List<PostModel>? postModels;
  List<UserModel>? members;

  GroupModel(
      {this.name,
      this.id,
      this.admin,
      this.postModels,
      this.members,
      this.imgPath,
      this.createdAt,
      this.description});

  GroupModel.fromJson(Map<String, dynamic> json) {
    id = json["id"].toString();
    name = json["name"];
    description = json["description"];
    createdAt = DateTime.parse(json["createdAt"]);
    if (json["imgPath"] != null) imgPath = json["imgPath"];
    if (json["admin"] != null) admin = UserModel.fromJson(json["admin"]);
    if (json['members'] != null) {
      members = <UserModel>[];
      json['members'].forEach((v) {
        members!.add(UserModel.fromJson(v));
      });
    }
    if (json['postModels'] != null) {
      postModels = <PostModel>[];
      json['postModels'].forEach((v) {
        postModels!.add(PostModel.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['name'] = name;
    data['description'] = description;
    data['createdAt'] = createdAt.toString();
    if (imgPath != null) data['imgPath'] = imgPath;
    if (admin != null) data['admin'] = admin!.toJson();
    if (members != null) {
      data['members'] = members!.map((v) => v.toJson()).toList();
    }
    if (postModels != null) {
      data['postModels'] = postModels!.map((v) => v.toJson()).toList();
    }
    return data;
  }

  get getId => id;

  set setId(String id) => this.id = id;

  get getName => name;

  set setName(name) => this.name = name;

  get getAdmin => admin;

  set setAdmin(admin) => this.admin = admin;

  get getImage => imgPath;

  set setImage(imgPath) => this.imgPath = imgPath;

  get getCreatedAt => createdAt;

  set setCreatedAt(createdAt) => this.createdAt = createdAt;

  get getDescription => description;

  set setDescription(description) => this.description = description;

  get getPostModels => postModels;

  set setPostModels(postModels) => this.postModels = postModels;

  get getMembers => members;

  set setMembers(members) => this.members = members;

  static List<GroupModel> groups = [
    GroupModel(
        id: "1",
        name: "drama news",
        admin: MyApp.currentUser,
        members: UserModel.users,
        imgPath: "images/photo1.jpg"),
    GroupModel(
        id: "2",
        name: "drama news",
        admin: MyApp.currentUser,
        members: UserModel.users,
        imgPath: "images/photo1.jpg"),
    GroupModel(
        id: "3",
        name: "drama news",
        admin: MyApp.currentUser,
        members: UserModel.users,
        imgPath: "images/photo1.jpg"),
    GroupModel(
        id: "4",
        name: "drama news",
        admin: MyApp.currentUser,
        members: UserModel.users,
        imgPath: "images/photo1.jpg"),
    GroupModel(
        id: "5",
        name: "drama news",
        admin: UserModel.users[3],
        members: UserModel.users,
        imgPath: "images/photo1.jpg"),
    GroupModel(
        id: "6",
        name: "drama news",
        admin: MyApp.currentUser,
        members: UserModel.users,
        imgPath: null),
    GroupModel(
        id: "7",
        name: "drama news",
        admin: UserModel.users[3],
        members: UserModel.users,
        imgPath: "images/photo1.jpg"),
    GroupModel(
        id: "8",
        name: "drama news",
        admin: UserModel.users[3],
        members: UserModel.users,
        imgPath: "images/photo1.jpg"),
  ];
}
